### PR TITLE
feat(shorebird_cli): Add support for updater_tools

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds the updater_tools artifact. Currently, this is just a wrapper around the patch executable.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore
- [ ] 🧪 Tests

### CODE REVIEW

File: packages/shorebird_cli/lib/src/executables/aot_tools.dart [hunk 1]
```diff
@@ -169,7 +169,7 @@ class AotTools {
       workingDirectory: workingDirectory,
     );
 
-    if (result.exitCode != 0) {
+    if (result.exitCode != ExitCode.success.code) {
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/executables/aot_tools.dart:172]: Some future patch probably makes this result.isSuccess

File: packages/shorebird_cli/lib/src/artifact_manager.dart [hunk 2]
```diff
@@ -40,12 +40,19 @@ class ArtifactManager {
 
     final tempDir = await Directory.systemTemp.createTemp();
     final diffPath = p.join(tempDir.path, 'diff.patch');
-
-    await patchExecutable.run(
-      releaseArtifactPath: releaseArtifactPath,
-      patchArtifactPath: patchArtifactPath,
-      diffPath: diffPath,
-    );
+    if (File(updaterTools.path).existsSync()) {
```
Comment (erickzanardo) [inline packages/shorebird_cli/lib/src/artifact_manager.dart]: Super minor, opitinal, nit, can we extract this `File` to a variable?

File: packages/shorebird_cli/lib/src/executables/updater_tools.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    );
+  }
+
+  /// Create a binary diff between a release artifact and a patch artifact.
+  Future<void> createDiff({
+    required String releaseArtifactPath,
+    required String patchArtifactPath,
+    required String diffPath,
+  }) async {
+    if (!File(releaseArtifactPath).existsSync()) {
```
Comment (erickzanardo) [inline packages/shorebird_cli/lib/src/executables/updater_tools.dart]: Same nit here and the on below

File: packages/shorebird_cli/lib/src/shorebird_artifacts.dart [hunk 4]
```diff
@@ -18,6 +18,9 @@ enum ShorebirdArtifact {
 
   /// The gen_snapshot executable.
   genSnapshot,
+
+  /// TODO
```
Comment (erickzanardo) [inline packages/shorebird_cli/lib/src/shorebird_artifacts.dart]:
```suggestion
  /// The updater tools executable
```

Or something like that

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/shorebird_artifacts.dart]: Done

File: packages/shorebird_cli/lib/src/artifact_manager.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/artifact_manager.dart]: Updated the function to accept Files, so this is not longer valid.

File: packages/shorebird_cli/lib/src/executables/updater_tools.dart (hunk 3 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/executables/updater_tools.dart]: Updated the function to accept Files, so this is not longer valid.

File: packages/shorebird_cli/lib/src/artifact_manager.dart (hunk 2 above)
Comment (erickzanardo) [reply packages/shorebird_cli/lib/src/artifact_manager.dart]: But if `updaterTools` is a file itself, can't we just do `updaterTools.existsSync()` instead?

**Review by eseidel — APPROVED** [review]

**Pull request merged**